src/user/provisioners.rs

```diff
@@ -121,46 +103,36 @@ impl Provisioners {
     /// Adds a new member with reward=0 and elibile_since=0.
     ///
     /// Useful for implementing unit tests.
-    pub fn add_member_with_value(&mut self, pubkey_bls: PublicKey, value: u64) {
+    pub fn add_member_with_value(&mut self, pubkey_bls: ConsensusPublicKey, value: u64) {
         self.add_member_with_stake(pubkey_bls, Stake::new(value, 0, 0));
     }
 
     /// Turns on/off elibility flag of stakes for a given round.
     pub fn update_eligibility_flag(&mut self, round: u64) {
-        for m in self.members.iter_mut() {
-            m.1.update_eligibility_flag(round)
-        }
+        self.members
+            .values_mut()
+            .for_each(|m| m.update_eligibility_flag(round));
     }
 
     /// Returns number of provisioners that owns at least one eligibile stake.
     pub fn get_eligible_size(&self, max_size: usize) -> usize {
-        let mut size = 0;
-        for (_, member) in self.members.iter() {
-            for (_, eligible) in member.stakes.iter() {
-                if *eligible {
-                    size += 1;
-                    break;
-                }
-            }
-
-            if size >= max_size {
-                return max_size;
-            }
-        }
-
-        size
+        self.members
+            .iter()
+            .filter(|(_, m)| m.stakes.iter().any(|(_, elegible)| *elegible))
+            .count()
```

Comment:
Keep in mind that `get_eligible_size` is called with `max_size = 1` anytime we execute sortition for Selection step.

This impl consumes the iterator and loop through all existing provisioners. Even this looks more idiomatic, I'd like to stick with the former (more efficient) implementation where we return once the `max_size` is reached.

Reply:
Changed to behave like the previous implementation